### PR TITLE
add more parallelism to nightly builds

### PR DESCRIPTION
**Pull request by arroyc**
I recreated the PR because I was getting numerous conflicts while rebasing on master, so I had to  squash all my commits in a single commit and then it was easier for me rebase .. thanks for the idea @kichalla .. But in that process I had to sacrifice my remote branch

### CODE REVIEW

File: vsts/pipelines/nightly.yml [hunk 1]
```diff
@@ -48,8 +48,8 @@ jobs:
     displayName: 'Set variables'
   - template: _buildParallel.yml
 
-- job: Job_IntegrationTests
-  displayName: Test Integration
+- job: Job_PythonIntegrationTests
```
Comment (kichalla) [inline vsts/pipelines/nightly.yml]: Basically you can have a template for a `job` for integration tests itself (related to what you mentioned in scrum meeting today)...job templates can be passed the parameters too

Comment (arroyc) [reply vsts/pipelines/nightly.yml]: yes, similarly I want have template for build image and test image etc etc

File: tests/Oryx.Tests.Common/DockerCli.cs [hunk 2]
```diff
@@ -18,6 +18,8 @@ public class DockerCli
 
         private readonly TimeSpan _waitTimeForExit;
         private readonly IEnumerable<EnvironmentVariable> _globalEnvVars;
+        private readonly Object Lock1 = new Object();
```
Comment (gpcastro) [inline tests/Oryx.Tests.Common/DockerCli.cs]: let's use more meaningful names, e.g. "outputLogLock"?

File: tests/Oryx.Tests.Common/DockerCli.cs [hunk 3]
```diff
@@ -134,11 +136,17 @@ public DockerCli(TimeSpan waitTimeForExit, IEnumerable<EnvironmentVariable> glob
                     // are called for each line that is written to the output.
                     standardOutputHandler: (sender, args) =>
                     {
-                        outputBuilder.AppendLine(args.Data);
+                        lock (Lock1)
```
Comment (gpcastro) [inline tests/Oryx.Tests.Common/DockerCli.cs]: as we discussed, let's put the locks in the same scope as the object it is protecting

File: vsts/scripts/pullandretag.sh [hunk 4]
```diff
@@ -5,23 +5,61 @@
 # --------------------------------------------------------------------------------------------
 
 set -o pipefail
+# $1 > buildimage-acr.txt
+# $2 > runtime-images-acr.txt
+declare integrationtestfilter="oryxdevmcr.azurecr.io/public/oryx"
```
Comment (gpcastro) [inline vsts/scripts/pullandretag.sh]: I would use a different name here, e.g. `imageFilter`?

File: tests/Oryx.Tests.Common/DockerCli.cs (hunk 3 above)
Comment (arroyc) [reply tests/Oryx.Tests.Common/DockerCli.cs]: reverted lock

File: tests/Oryx.Tests.Common/DockerCli.cs (hunk 2 above)
Comment (arroyc) [reply tests/Oryx.Tests.Common/DockerCli.cs]: reverted lock

**Pull request merged**